src/bundle/Resources/public/js/scripts/core/dropdown.js

```diff
@@ -250,6 +260,30 @@
             }
         }
 
+        filterItems(event) {
+            const allItems = this.itemsListContainer.querySelectorAll('[data-filter-value]');
+
+            if (event.currentTarget.value.length < 3) {
+                [...allItems].forEach((item) => item.classList.remove(CLASS_ITEM_HIDDEN));
+
+                return;
+            }
+
+            const visibleItems = this.itemsListContainer.querySelectorAll(`[data-filter-value^="${event.currentTarget.value.toLowerCase()}"]`);
+
+            [...allItems].forEach((item) => item.classList.add(CLASS_ITEM_HIDDEN));
+            [...visibleItems].forEach((item) => item.classList.remove(CLASS_ITEM_HIDDEN));
```

Comment:
First it hides all items and in next loop shows matched items.
It may cause some visible artifacts. Can we do it in one loop and check if it should be visible or not inside?